tests/integration_tests/test_uv_functionality.py

```diff
@@ -0,0 +1,94 @@
+"""Tests for UV functionality and integration validation."""
+
+from __future__ import annotations
+
+import os
+import subprocess
+import tempfile
+from pathlib import Path
+
+import pytest
+
+
+def test_uv_available() -> None:
+    """Test that UV is available on the command line."""
+    # Use poetry's virtual environment where UV is installed
+    result = subprocess.run(
+        ["poetry", "run", "uv", "--version"],
+        capture_output=True,
+        text=True,
+        env=os.environ.copy(),  # Ensure we have the poetry environment
+    )
+    assert result.returncode == 0
+    assert "uv" in result.stdout.lower()
+
+
+@pytest.fixture
+def temp_venv() -> Path:
+    """Create a temporary directory for virtual environment testing."""
```

Comment:
Watch this on Windows CI to ensure we don't fail on exciting the context handler.